File: lumina_beauty_final_version/lumina_beauty_backend/LuminaBeauty-Model/src/main/java/pe/edu/pucp/luminaBeauty/Model/DetallePedido.java
package pe.edu.pucp.luminaBeauty.Model;
import java.math.BigDecimal;

public class DetallePedido {
    private int id_detalle_pedido;
    private String nombre_producto;
    private String sku_producto;
    private int cantidad;
    private BigDecimal precio_unitario;

    private Pedido pedido;
    private Producto producto;

    public DetallePedido() {
    }

    public int getId_detalle_pedido() {
        return id_detalle_pedido;
    }

    public void setId_detalle_pedido(int id_detalle_pedido) {
        this.id_detalle_pedido = id_detalle_pedido;
    }

    public String getNombre_producto() {
        return nombre_producto;
    }

    public void setNombre_producto(String nombre_producto) {
        this.nombre_producto = nombre_producto;
    }

    public String getSku_producto() {
        return sku_producto;
    }

    public void setSku_producto(String sku_producto) {
        this.sku_producto = sku_producto;
    }

    public int getCantidad() {
        return cantidad;
    }

    public void setCantidad(int cantidad) {
        this.cantidad = cantidad;
    }

    public BigDecimal getPrecioUnitario() {
        return precio_unitario;
    }

    public void setPrecioUnitario(BigDecimal precio_unitario) {
        this.precio_unitario = precio_unitario;
    }

    public Pedido getPedido() {
        return pedido;
    }

    public void setPedido(Pedido pedido) {
        this.pedido = pedido;
    }

    public Producto getProducto() {
        return producto;
    }

    public void setProducto(Producto producto) {
        this.producto = producto;
    }
}

File: lumina_beauty_final_version/lumina_beauty_frontend/LuminaBeauty/Models/Orders/DetallePedido.cs
using System.Text.Json.Serialization;

namespace LuminaBeauty.Models
{
    public class DetallePedido
    {
        [JsonPropertyName("id_detalle_pedido")]
        public int IdDetallePedido { get; set; }

        [JsonPropertyName("nombre_producto")]
        public string NombreProducto { get; set; } = string.Empty;

        [JsonPropertyName("sku_producto")]
        public string SkuProducto { get; set; } = string.Empty;

        [JsonPropertyName("producto")]
        public Producto? Producto { get; set; }

        [JsonPropertyName("cantidad")]
        public int Cantidad { get; set; }

        [JsonPropertyName("precioUnitario")]
        public decimal PrecioUnitario { get; set; }

        private decimal _subtotal;

        [JsonPropertyName("subtotal")]
        public decimal Subtotal
        {
            get => _subtotal > 0 ? _subtotal : PrecioUnitario * Cantidad;
            set => _subtotal = value;
        }
    }
}